flow.py

```diff
@@ -131,45 +277,123 @@ def receive_ack(self, packet):
         assert packet.src == original_packet.dst and packet.dst == original_packet.src
         rtt = globals_.event_manager.get_time() - time_sent
         self.rtte.update_rtt_datapoint(rtt)
-        del self.packets_waiting_for_acks[packet.id_]
+
+        # del self.packets_waiting_for_acks[packet.id_]
+        self.update_packets_waiting_for_ack(packet.next_expected)
         globals_.stats_manager.notify(self.rt_packet_delay_graph_tag, rtt)
 
+    def reset_timeout(self, packet):
```

Comment:
Is reset_timeout() being called anywhere? Will it be called anywhere? It looks like the only call to it is commented out